app/services/git/fetches_tracks.rb

```diff
@@ -1,25 +1,26 @@
 class Git::FetchesTracks
 
-  QUIET_PERIOD = 30.seconds
   ERROR_BACKOFF_PERIOD = 10.seconds
 
-  def self.run
-    new.run
+  def self.fetch(tracks)
+    new(tracks).fetch
   end
 
-  def run
-    loop do
-      fetch_problem_specs
-      fetch_website_content
-      Track.find_each do |track|
-        fetch(track)
-        sleep 1.second
-      end
-      sleep QUIET_PERIOD
+  def initialize(tracks)
+    @tracks = tracks
+  end
+
+  def fetch
+    fetch_problem_specs
+    fetch_website_content
+    tracks.each do |track|
+      fetch_track(track)
+      sleep 1.second
```

Comment:
This my code not yours, but I should probably have extracted the backoff into a constant.

Reply:
No problem! I'll extract that to a constant.